kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -33,6 +33,11 @@
 public final class DeltaErrors {
   private DeltaErrors() {}
 
+  public static KernelException missingCheckpoint(String tablePath, long checkpointVersion) {
+    return new InvalidTableException(
+        tablePath, String.format("Missing checkpoint at version %s", checkpointVersion));
```

Comment:
Hmm I feel like this message lacks context... why are we missing a checkpoint / why did we expect one to be there?

Without this, I think it makes you wonder is it missing because we truncated the log (i.e. version not reconstructable) which is a different error case, or something else?

Reply:
I agree with this. I think plumbing that context through the various SnapshotManager APIs is out of scope for this refactor/consolidation PR. Unfortunately, I think our best path forwards here is to live with this current state in this PR and refactor it in a follow up.